src/pseudo/NormConservingPsp.jl

```diff
@@ -66,6 +70,12 @@ charge, which is assumed here.
 
 Notice: The returned result is the *energy per unit cell* and not the energy per volume.
 To obtain the latter, the caller needs to divide by the unit cell volume.
+
+he energy correction is defined as the limit of the Fourier-transform of the local
+potential as q -> 0, using the same correction as in the Fourier-transform of the local
+potential:
+lim{q->0} 4π Nelec ∫{R+} (V(r) - C(r)) sin(qr)/qr r^2 dr + F[C(r)]
+= 4π Nelec ∫{R+} (V(r) + Z/r) r^2 dr
 """
 eval_psp_energy_correction(T, psp::NormConservingPsp, n_electrons) = zero(T)
```

Comment:
I wonder whether we should give a better fallback here.

Reply:
error() is better :-) I think it dates back from when we (at least I) thought the energy correction was just an arbitrary number added to the system, rather than something from electrostatics. Couldn't we just take the above formula using the potential at sqrt(eps()) or something? We can also use it as a nice test

Reply:
I have an idea for this that I'm working on [here](https://github.com/azadoks/PseudoPotentialIO.jl/tree/interface).

If the interface provides functions to get at `V(r)` (as a Julia function, e.g. an interpolator) and `Z_{valence}`, we can write a meaningful fallback for this that "just works" if the other two (very simple) functions are (properly) implemented and errors otherwise. It will likely be painfully slow, but it's only really there as an example and "canonical" definition.

Otherwise, yes, just `error()`.

Reply:
If it's not a one-liner I wouldn't bother too much, it's not like we're going to implement a new pseudo format every day (hopefully). This is however a great test, and you can just refer to the test in the source.